File: Casks/brewui.rb
cask "brewui" do
  version "0.4.2"
  sha256 "54bbbfe849256b7e4747379510a20c6badf8df43818d60b83feb630a55ed8c91"

  # 镜像到本仓 Release（tag brewui-<ver>，资产名沿用上游）：
  # 本 tap 政策是所有 cask 都镜像最新版（直引只做过渡）；
  # url 用 #{version} 插值，否则 audit 会因"URL 无版本"要求 sha256 :no_check。
  # 版本更新由 updater/brewui.swift 跟 GitHub release（UpdaterCore github 流），
  # 有更新时下载、上传 Release、改写本文件。
  url "https://github.com/Bemly/homebrew-tahoe/releases/download/brewui-0.4.2/Homebrew-0.4.2.zip"
  name "BrewUI"
  desc "Official graphical interface for Homebrew"
  homepage "https://github.com/Homebrew/BrewUI"

  # 本 tap 只收录 macOS 26(Tahoe) 及以上可用的软件。
  # Universal 包（x86_64+arm64 双切片，实测），单包覆盖双架构，无需 arch 分包。
  depends_on macos: :tahoe

  app "Homebrew.app"

  uninstall quit: "sh.brew.app"
end

File: updater/brewui.swift
// updater/brewui.swift —— 与 Casks/brewui.rb 一一同名的检查器入口
//
// BrewUI（Homebrew 官方图形界面）不在 homebrew/core，走 UpdaterCore 的
// github 流：版本判据是 releases/latest 的跳转目标 tag（HEAD 取 Location，
// 不消耗 GitHub API 限额），tag 前缀 "v" 剥离后即版本号。
//
// 上游 release 只有 dSYMs/pkg/zip 三个资产、无 checksums 文件 → 有更新时
// 回退下载 zip 本地实算（约 8MB，仅检测到新版本时发生），并按本 tap 政策
// （所有 cask 都镜像最新版）上传本仓 Release（tag brewui-<ver>），cask
// url 指 Release（workbuddy 同机制）。
// cask 无 bottle 机制，watcher 更新 cask 后不触发 bottle.yml。
//
// 注意：Swift 要求实参顺序与 CheckConfig.init 的形参声明一致。

import Foundation

@main
struct BrewuiCheck {
    static func main() {
        runCheck(CheckConfig(
            formula: "brewui",
            formulaPath: "Casks",
            isCask: true,
            downloadURL: { version in
                "https://github.com/Homebrew/BrewUI/releases/download/v\(version)/Homebrew-\(version).zip"
            },
            checksumsURL: nil,
            uploadRelease: true,
            githubRepo: "Homebrew/BrewUI"
        ))
    }
}
